pycytominer/cyto_utils/load.py

```diff
@@ -87,16 +148,40 @@ def load_profiles(profiles):
     FileNotFoundError
         Raised if the provided profile does not exists
     """
-    if not isinstance(profiles, pd.DataFrame):
-        # Check if path exists and load depending on file type
-        if is_path_a_parquet_file(profiles):
-            return pd.read_parquet(profiles, engine="pyarrow")
 
+    # If already a dataframe, return it
+    if isinstance(profiles, pd.DataFrame):
+        return profiles
+
+    # Check if path exists and load depending on file type
+    if is_path_a_parquet_file(profiles):
+        return pd.read_parquet(profiles, engine="pyarrow")
+
+    # Check if path is an AnnData file
+    anndata, anndata_type = is_anndata(profiles)
+    if anndata:
+        if anndata_type == "h5ad":
+            adata = ad.read_h5ad(profiles, backed="r")
+        elif anndata_type == "zarr":
+            adata = ad.read_zarr(profiles)
+        elif anndata_type == "in-memory":
+            adata = profiles
+        else:
+            raise ValueError("Unrecognized AnnData type")
+
+        # Convert to dataframe
+        if adata.isbacked:
+            # if we're backed by a file, just read it directly
```

Comment:
this is a curiosity question, but could pycytominer work on backed data? WOuld this have speed up benefits? (probably worth discussing in the context of a `narwhals` upgrade in 2.0 as well)

Reply:
It could work with backed data / lazy approached. However, it will come with a great deal of technical complexity no matter which API we choose (we won't be able to assume the materialization of the data as we do with Pandas currently). For example, `sklearn` methods might not all be "lazy compatible" or allow for partial fits.

It might be that we could bridge from backed `anndata` to a lazy frame of some kind with `polars`. I'd recommend opting for an abstraction layer such as `narwhals` or similar for internal processing so we leave the door open for current and future code development.

Reply:
Opened #576 in reference to this.